internal/provider/alert_resource.go

```diff
@@ -140,8 +138,10 @@ func (r *AlertResource[T]) Delete(ctx context.Context, req resource.DeleteReques
 		return
 	}
 
-	if os.Getenv("DEBUG_ALERT") == "1" {
-		fmt.Printf("----- Delete Alert ----- %+v\n", alert)
+	if os.Getenv("TF_LOG_PROVIDER") == "TRACE" {
+		msg := Json("----- Delete Alert ---", alert)
+		tflog.Trace(ctx, msg)
+		fmt.Println(msg)
```

Comment:
suggestion: there is no need for the env check, just use tflog.Trace() and you can toggle the logging level per provider like this: `TF_LOG_PROVIDER_MEZMO=TRACE` 

```suggestion
        tflog.Trace(ctx, Json("----- Delete Alert ---", alert))
```

Reply:
The Env check is specifically for printiln. Tflog doesn't work during testing.

Reply:
It does, you just need to pass -v to `go test`

Reply:
That's something I didn't know--tacking on the provider name to the log call? `TF_LOG_PROVIDER_MEZMO`?  I was only aware of the overall env var `TF_LOG_PROVIDER=TRACE`

In any case, when I remove the println, I either get nothing (when using "_MEZMO"), or I don't get the traces.  Maybe this is a TF version issue.  What are you using?  I'm on 1.5.7.

```
λ ➜ npm run local -- TF_LOG_PROVIDER_MEZMO=TRACE TF_ACC=1 go test -v -run 'Test*' ./internal/provider/models/processors/test

> terraform-provider-mezmo@0.0.0 local
> env $(cat ./env/local.env) TF_LOG_PROVIDER_MEZMO=TRACE TF_ACC=1 go test -v -run Test* ./internal/provider/models/processors/test

=== RUN   TestAggregateV2Processor
    utils.go:106: Created account for testing 
--- PASS: TestAggregateV2Processor (5.40s)
=== RUN   TestCompactFieldsProcessor
--- PASS: TestCompactFieldsProcessor (2.14s)
=== RUN   TestDecryptFieldsProcessor
--- PASS: TestDecryptFieldsProcessor (2.34s)
```

And with this, I don't see traces for the requests which have the println commented out for processor reqeusts.  We see the results, but not the requests.
```
λ ➜ npm run local -- TF_LOG_PROVIDER=TRACE go test -v -run 'Test*' ./internal/provider/models/processors/test

> terraform-provider-mezmo@0.0.0 local
> env $(cat ./env/local.env) TF_LOG_PROVIDER=TRACE go test -v -run Test* ./internal/provider/models/processors/test

=== RUN   TestAggregateV2Processor
    utils.go:106: Created account for testing 
Result of POST http://localhost:19095/v3/pipeline {
  "data": {
    "id": "144ce734-2fdb-11ef-838e-0242ac130004",
    "title": "pipeline",
    "created_at": "2024-06-21T14:32:25.669Z"
  }
}
----- Destination TO Create api --- {
  "type": "http",
  "user_config": {
    "capture_metadata": false,
    "decoding": "auto"
  },
  "generation_id": 0
}
Result of POST http://localhost:19095/v3/pipeline/144ce734-2fdb-11ef-838e-0242ac130004/transform {
  "data": {
    "id": "144f4f24-2fdb-11ef-838e-0242ac130004",
    "type": "aggregate-v2",
    "title": "My aggregate v2 processor",
    "description": "Lets aggregate stuff",
    "user_config": {
      "evaluate": {
        "operation": "SUM"
      },
      "event_timestamp": ".timestamp",
      "window": {
        "interval": 3600,
        "type": "tumbling"
      }
    },
    "generation_id": 0,
    "outputs": [
      {
        "id": "144f4f24-2fdb-11ef-838e-0242ac130004",
        "label": "Default"
      }
    ]
  }
}
Result of POST http://localhost:19095/v3/pipeline/144ce734-2fdb-11ef-838e-0242ac130004/source {
  "data": {
    "id": "14501ba2-2fdb-11ef-9b30-0242ac130004",
    "type": "http",
    "user_config": {
      "capture_metadata": false,
      "decoding": "auto"
    },
    "generation_id": 0,
    "gateway_route_id": "14501ba2-2fdb-11ef-9b30-0242ac130004"
  }
}

```

Reply:
I'm on latest now. Besides removing the fmt.Println you also have to remove the if statement, otherwise the tflog call wont be made. 

```
$ terraform --version
Terraform v1.8.5
on darwin_arm64
+ provider registry.terraform.io/mezmo/mezmo v3.0.2
```

```diff
$ git diff --no-ext-diff |cat
diff --git a/internal/client/client.go b/internal/client/client.go
index 6053402..8e93e3e 100644
--- a/internal/client/client.go
+++ b/internal/client/client.go
@@ -6,7 +6,6 @@ import (
        "fmt"
        "io"
        "net/http"
-       "os"
 
        "github.com/hashicorp/terraform-plugin-log/tflog"
        . "github.com/mezmo/terraform-provider-mezmo/internal/provider/models/modelutils"
@@ -93,11 +92,8 @@ func readBody(result any, resp *http.Response, ctx context.Context) error {
                return nil
        }
        err = json.Unmarshal(bodyBuffer, result)
-       if err == nil && os.Getenv("TF_LOG_PROVIDER") == "TRACE" {
-               msg := Json(fmt.Sprintf("Result of %s %s", resp.Request.Method, resp.Request.URL), result)
-               tflog.Trace(ctx, msg)
-               fmt.Println(msg)
-       }
+       msg := Json(fmt.Sprintf("Result of %s %s", resp.Request.Method, resp.Request.URL), result)
+       tflog.Trace(ctx, msg)
        return err
 }
```

Reply:
lol, doh!  Fixed.

I also moved all the logging statements into client.go which cleaned up the scattered nature of where the logging statements existed.